plugins/module_utils/rest_client.py

```diff
@@ -112,28 +113,30 @@ def delete_record(
     def put_record(
         self,
         endpoint: str,
-        payload: dict[Any, Any],
+        payload: Optional[dict[Any, Any]],
         check_mode: bool,
         query: Optional[dict[Any, Any]] = None,
         timeout: Optional[float] = None,
-        binary_data: Optional[bytes] = None,
+        binary_data: Optional[Union[bytes, BufferedReader]] = None,
         headers: Optional[dict[Any, Any]] = None,
     ) -> TypedTaskTag:
-        # Method put doesn't support check mode # IT ACTUALLY DOES
         if check_mode:
             return utils.MOCKED_TASK_TAG
         try:
-            response: TypedTaskTag = self.client.put(
+            tt = self.client.put(
                 endpoint,
                 data=payload,
                 query=query,
                 timeout=timeout,
                 binary_data=binary_data,
                 headers=headers,
-            ).json
+            )
+            try:
+                response: TypedTaskTag = tt.json
+            except Exception:
```

Comment:
No. I pushed 'TEMP TEMP ..." commit just to show the `.json` part is problem. But we cannot catch arbitrary exception and essentially ignore it.

Who is calling this code? Maybe we need to let caller decide if conversion to json is possible? Or just catch JsonDecodeError in virtual_disk.py? Can you in this case be sure all data was uploaded - I think yes.

Reply:
Oh I thought this was a part of temporary ISO bug fix, so I just copy pasted your example.